src/OpenStudy/study_openinterface.jl

```diff
@@ -768,8 +822,34 @@ function configuration_parameter(
     return out
 end
 
-_cast(::Type{T}, val::T) where T = val
-function _cast(::Type{T}, val::String) where T
+"""
+    _cast(::Type{T}, val, default::T = _default_value(T))
+
+Converts `val` to type `T`, if possible.
+"""
+_cast(::Type{T}, val::T, default::T = _default_value(T)) where T = val
+_cast(::Type{String}, val::String, default::String = _default_value(String)) = val
+_cast(::Type{Int32}, val::Integer, default::Int32 = _default_value(Int32)) = Int32(val)
+_cast(::Type{Float64}, val::Float64, default::Float64 = _default_value(Float64)) = val
+_cast(::Type{Dates.Date}, val::Dates.Date, default::Dates.Date = _default_value(Dates.Date)) = val
```

Comment:
We don't even use the argument `default` in these functions.

Reply:
We use in the nothing case so the functions need to have it as a positional arg.